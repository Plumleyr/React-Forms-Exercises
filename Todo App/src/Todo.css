.Todo-todo{
  display: flex;
  align-items: center;
  text-align: center;
}

.Todo-btn{
  background-color: lightblue;
  margin-left: 10px;
  height: 24px;
  width: 24px;
  border-radius: 10px;
  border: 1px solid white;
}

.Todo-btn:hover{
  background-color: lightcoral;
}