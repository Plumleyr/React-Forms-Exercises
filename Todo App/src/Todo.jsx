import './Todo.css'

const Todo = ({todo, id, removeTodo, "data-testid": testId,}) => {
  return(
    <div className="Todo-todo" data-testid={testId}>
      <p>{todo}</p>
      <button className='Todo-btn' onClick={() => removeTodo(id)}>x</button>
    </div>
  )
}

export default Todo