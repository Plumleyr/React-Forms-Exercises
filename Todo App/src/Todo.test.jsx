import {render} from '@testing-library/react'
import Todo from './Todo'
import { expect, it } from 'vitest'

it('should render without crashing', () => {
  render(<Todo todo={'go swimming'}/>);
});

it('should match snapshot', () => {
  const {asFragment} = render(<Todo todo={'go swimming'}/>);
  expect(asFragment()).toMatchSnapshot();
});