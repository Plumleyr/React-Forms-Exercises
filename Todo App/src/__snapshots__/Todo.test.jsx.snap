// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo-todo"
  >
    <p>
      go swimming
    </p>
    <button
      class="Todo-btn"
    >
      x
    </button>
  </div>
</DocumentFragment>
`;
